Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py

```diff
@@ -208,30 +236,37 @@ def instance_info(instance_id: str, public_ip: str) -> Dict:
         public_ip (str): Public IP address of the EC2 instance
 
     Returns:
-        Dict: interface to security group mapping
+        mapping_dict (Dict): interface to security group mapping
+        integration_to_use (str)
     """
-    instance_info = demisto.executeCommand("aws-ec2-describe-instances", {"instanceIds": instance_id})
-    if instance_info[0].get('Contents').get('AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces'):
-        interfaces = instance_info[0].get('Contents').get(
-            'AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces')
-        match = False
-        mapping_dict = {}
-        for interface in interfaces:
-            if interface.get('Association'):
-                if interface.get('Association').get('PublicIp') == public_ip:
+    cmd_args = {"instanceIds": instance_id}
+    if assume_role:
+        cmd_args.update({'roleArn': assume_role})
+        cmd_args.update({'roleSessionName': "test"})
+    instance_info = demisto.executeCommand("aws-ec2-describe-instances", cmd_args)
+    # Need a for loop in case multiple AWS-EC2 integrations are configured.
+    match = False
+    for instance in instance_info:
+        # Check if returned error, in the case of multiple integration instances only one should pass.
+        if not isError(instance) and \
+           instance.get('Contents').get('AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces'):
+            interfaces = instance.get('Contents').get(
+                'AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces')
+            mapping_dict = {}
+            for interface in interfaces:
+                if interface.get('Association') and interface.get('Association').get('PublicIp') == public_ip:
                     match = True
                     group_list = []
                     for sg in interface['Groups']:
                         group_list.append(sg['GroupId'])
                     mapping_dict[interface['NetworkInterfaceId']] = group_list
-        if match is False:
-            raise ValueError('could not find interface with public IP association')
-        return mapping_dict
-    else:
-        raise ValueError('failed to pull information on EC2 instance')
+                    integration_to_use = instance['ModuleName']
```

Comment:
Let's add a `break` here. No need to continue looping if we found our instance.
(may have messed up the correct indentation in this suggestion, probably better to add this line in an IDE)
```suggestion
                    integration_to_use = instance['ModuleName']
            break
```

Reply:
change in [review updates](https://github.com/demisto/content/pull/29311/commits/d0b01a94de4508529522275932721c124c094d62)

Reply:
This will only break from the internal interfaces loop, but not the`for instance in instance_info` loop (it will keep searching the other instances even when a match is already found)

Something like this should work (my previous suggestion wasn incorrect):
```python
    for instance in instance_info:
        # Check if returned error, in the case of multiple integration instances only one should pass.
        if not isError(instance) and \
           instance.get('Contents').get('AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces'):
            interfaces = instance.get('Contents').get(
                'AWS.EC2.Instances(val.InstanceId === obj.InstanceId)')[0].get('NetworkInterfaces')
            mapping_dict = {}
            for interface in interfaces:
                if interface.get('Association') and interface.get('Association').get('PublicIp') == public_ip:
                    match = True
                    group_list = []
                    for sg in interface['Groups']:
                        group_list.append(sg['GroupId'])
                    mapping_dict[interface['NetworkInterfaceId']] = group_list
                    instance_to_use = instance['ModuleName']
                    break  # <- What's already added

        if match:  # <- I've added this
            break

    if match is False:
        raise ValueError('could not find interface with public IP association')

```

Reply:
change made in [update 9/7](https://github.com/demisto/content/pull/29311/commits/cea008683bf63abf4ae2c6a3f1ece7795b5b5e46)